sklearn/notebooks/hp_tuning_auto_mpg/ml_engine_hp_tuning_training.ipynb

```diff
@@ -0,0 +1,907 @@
+      "  gcloud config set --help\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "! gcloud config set project $PROJECT_ID"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Submit the training job."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Job [auto_mpg_hp_tuning_20180912_103335] submitted successfully.\r\n",
+      "Your job is still active. You may view the status of your job with the command\r\n",
+      "\r\n",
+      "  $ gcloud ml-engine jobs describe auto_mpg_hp_tuning_20180912_103335\r\n",
+      "\r\n",
+      "or continue streaming the logs with the command\r\n",
+      "\r\n",
+      "  $ gcloud ml-engine jobs stream-logs auto_mpg_hp_tuning_20180912_103335\r\n",
+      "jobId: auto_mpg_hp_tuning_20180912_103335\r\n",
+      "state: QUEUED\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "! gcloud ml-engine jobs submit training auto_mpg_hp_tuning_$(date +\"%Y%m%d_%H%M%S\") \\\n",
+    "  --job-dir $JOB_DIR \\\n",
+    "  --package-path $TRAINER_PACKAGE_PATH \\\n",
+    "  --module-name $MAIN_TRAINER_MODULE \\\n",
+    "  --region $REGION \\\n",
+    "  --runtime-version=$RUNTIME_VERSION \\\n",
+    "  --python-version=$PYTHON_VERSION \\\n",
+    "  --scale-tier BASIC \\\n",
+    "  --config $HPTUNING_CONFIG"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# [Optional] StackDriver Logging\n",
+    "You can view the logs for your training job:\n",
+    "1. Go to https://console.cloud.google.com/\n",
+    "1. Select \"Logging\" in left-hand pane\n",
+    "1. Select \"Cloud ML Job\" resource from the drop-down\n",
+    "1. In filter by prefix, use the value of $JOB_NAME to view the logs\n",
+    "\n",
+    "On the logging page of your model, you can view the different results for each HP tuning job. \n",
+    "\n",
+    "Example:\n",
+    "```\n",
+    "{\n",
+    "  \"trialId\": \"2\",\n",
+    "  \"hyperparameters\": {\n",
+    "    \"selection\": \"random\",\n",
+    "    \"max_iter\": \"1892\",\n",
+    "    \"tol\": \"0.0609819896050862\",\n",
+    "    \"alpha\": \"4.3704164028167725\"\n",
+    "  },\n",
+    "  \"finalMetric\": {\n",
+    "    \"trainingStep\": \"1000\",\n",
+    "    \"objectiveValue\": 0.8658283435394591\n",
+    "  }\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# [Optional] Verify Model File in GCS\n",
+    "View the contents of the destination model folder to verify that all 30 model files have indeed been uploaded to GCS.\n",
+    "\n",
+    "Note: The model can take a few minutes to train and show up in GCS."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
```

Comment:
We probably want to make a uniform decision here: Should we commit a notebook before we run it (without any outputs), or after we run it (with the outputs displayed). I am leaning more toward a no-output commit. But I can be persuaded otherwise. Any thoughts here @dizcology ?

Reply:
I could go either way. I'll clear the output